ext/drx/drx.c

```diff
@@ -2787,27 +2867,31 @@ drx_event_restore_state(void *drcontext, bool restore_memory,
         /* Nothing to do if nobody had never called expand_scatter_gather() before. */
         return true;
     }
+    if (!info->fragment_info.app_code_consistent) {
+        /* Can't verify application code.
+         * XXX i#2985: is it better to keep searching?
+         */
+        return true;
+    }
     instr_init(drcontext, &inst);
     byte *pc = decode(drcontext, dr_fragment_app_pc(info->fragment_info.tag), &inst);
     if (pc != NULL) {
+        scatter_gather_info_t sg_info;
+        get_scatter_gather_info(&inst, &sg_info);
         if (instr_is_gather(&inst)) {
-            if (!info->fragment_info.app_code_consistent) {
```

Comment:
This looks like it was a bug: you were decoding even when it wasn't consistent.  I thought I pointed out that there was a missing check for being consistent in an earlier review: how did this check end up *after* the decode?  In any case it is in the right place now.

Reply:
Agreed.